eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -663,7 +655,7 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
           handleCommandSuccess(sender, d.copy(commitments = commitments1)) sending fulfill
         case Failure(cause) =>
           // we can clean up the command right away in case of failure
-          relayer ! CommandBuffer.CommandAck(d.channelId, c.id)
+          nodeParams.db.pendingRelay.removePendingRelay(d.channelId, c.id)
```

Comment:
I don't really like calling directly `pendingRelay.removePendingRelay` from here...

Reply:
> Because of the perf or because you'd like it to be hidden in a function?

Because of the encapsulation. Making a call to the db here isn't symmetrical with how the data was put in the db (which was hidden behind a `safeSend`. I think 1d4e4ef7b makes it better.